server/api/room.ts

```diff
@@ -355,67 +351,68 @@ const addToQueue: RequestHandler<
 	OttResponseBody<unknown>,
 	OttApiRequestAddToQueue
 > = async (req, res) => {
+	const body = OttApiRequestAddToQueueSchema.parse(req.body);
 	let points = 5;
-	if ("videos" in req.body) {
-		points = 3 * req.body.videos.length;
+	if ("videos" in body) {
+		points = 3 * body.videos.length;
 	}
 	if (!(await consumeRateLimitPoints(res, req.ip, points))) {
 		return;
 	}
 	const room = (await roommanager.getRoom(req.params.name)).unwrap();
 
-	let roomRequest: AddRequest;
-	if ("videos" in req.body) {
+	let roomRequest: AddRequest | undefined;
+	if ("videos" in body) {
 		roomRequest = {
 			type: RoomRequestType.AddRequest,
-			videos: req.body.videos,
+			videos: body.videos,
 		};
-	} else if ("url" in req.body) {
+	} else if ("url" in body) {
 		roomRequest = {
 			type: RoomRequestType.AddRequest,
-			url: req.body.url,
+			url: body.url,
 		};
-	} else if ("service" in req.body && "id" in req.body) {
+	} else if ("service" in body && "id" in body) {
 		roomRequest = {
 			type: RoomRequestType.AddRequest,
 			video: {
-				service: req.body.service,
-				id: req.body.id,
+				service: body.service,
+				id: body.id,
 			},
 		};
-	} else {
-		throw new BadApiArgumentException("service,id", "missing");
 	}
-	await room.processUnauthorizedRequest(roomRequest, { token: req.token! });
-	res.json({
-		success: true,
-	});
+
+	if (roomRequest !== undefined) {
+		await room.processUnauthorizedRequest(roomRequest, { token: req.token! });
+		res.json({
+			success: true,
+		});
+	}
```

Comment:
Which would make this check for undefined unnecessary.

The way this is now if there is a logic error and roomRequest is undefined, it will cause the server to never send a response.